brokerbot.go

```diff
@@ -120,95 +122,89 @@ func initTokens() {
 }
 
 func handleDefaultPort(w http.ResponseWriter, r *http.Request) {
-	log.Println("Heartbeat")
-	timeSinceLastHeartbeat = time.Now()
-	fmt.Fprintln(w, "Hello World!")
+	fmt.Fprintln(w, "OK")
 }
 
 func handleMessage(s *discordgo.Session, m *discordgo.MessageCreate) {
-	log.Printf("time since last heartbeat: %s", time.Since(timeSinceLastHeartbeat))
-	/* Validation */
 	if m.Author.ID == s.State.User.ID {
-		// Prevent the bot from talking to itself.
+		// Ignore messages from self.
 		return
 	}
 
-	isBotMention, msg := hasBotMention(s, m)
+	splitMsg := strings.Fields(m.ContentWithMentionsReplaced())
+
+	if splitMsg[0] != botHandle && splitMsg[0] != botPrefix {
+		// Message wasn't meant for us.
+		return
+	}
 
-	if !isBotMention {
-		// Ignore bot mentions.
+	if len(splitMsg) < 2 || splitMsg[1] == helpToken {
+		// Message didn't have enough parameters.
+		messagelib.SendMessage(s, m.ChannelID, getHelpMessage())
 		return
 	}
 
-	msg = strings.TrimSpace(msg)
-	userInput := strings.TrimPrefix(msg, "!stonks ")
-	userInput = strings.ToUpper(userInput)
-	expandedString := messagelib.ExpandAliases(userInput)
-	tickers := messagelib.DedupeTickerStrings(strings.Split(expandedString, " "))
+	var tickers []string = splitMsg[1:]
+	tickers = messagelib.RemoveMentions(tickers)
+	tickers = messagelib.CanonicalizeMessage(tickers)
+	tickers = messagelib.ExpandAliases(tickers)
+	tickers = messagelib.DedupeSlice(tickers)
+
+	startTime := time.Now()
+	log.Printf("Received request for tickers: %s", tickers)
+
+	log.Println("Fetching crypto price feeds.")
 
 	priceFeeds, err := cryptolib.GetPriceFeeds(geminiClient)
```

Comment:
Identifying areas of future work: Checking if we need to fetch PriceFeeds with the existence of "$" in the string.

Reply:
++